dojo/settings/settings.dist.py

```diff
@@ -255,7 +260,7 @@ def generate_url(scheme, double_slashes, user, password, host, port, path):
     'dojo.pipeline.social_uid',
     'social_core.pipeline.social_auth.auth_allowed',
     'social_core.pipeline.social_auth.social_user',
-    'social_core.pipeline.user.get_username',
```

Comment:
Removing this line broke the Google OAuth. Adding the pipeline back in did not mangle the username created by Azure as the associate_by_email pipeline.

Reply:
I had removed it because this is the line that was appending ranm string to the end of the user name, as documented here: https://python-social-auth.readthedocs.io/en/latest/pipeline.html#authentication-pipeline 

>#### Make up a username for this person, appends a random string at the end if there's any collision.
    'social_core.pipeline.user.get_username',

I have now found additional ways to [control username creation docmented in social-core](https://python-social-auth.readthedocs.io/en/latest/configuration/settings.html#username-generation) and will re-work based on it